### PR TITLE
The password input field is the wrong size

### LINKED ISSUE #449: The password input field is the wrong size
**Describe the bug**
When content is password protected, a password form is displayed.
The password input field is not the same height as the button, and the spacing between the field and the button is not 
the same if compared to for example the search form.

**To Reproduce**
1. Password protect a post or page, save.
2. View the front.
3. See that the input field and button does not have the same height.

**Expected behavior**
I think it would look better if the field and button sizes matched.
Perhaps the _Password:_ label should also be on its own line, above the input.

**Screenshots**

password-1

Password form overlapping the search form just to show the difference:
password

**Editor version (please complete the following information):**
- WordPress version: 5.5.1, 5.30.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome version 86.0.4240.75

### PR DESCRIPTION
**Pull request by SResok**
Fixes #449

## Summary

Added a the_password_form filter 
## Test instructions

This PR can be tested by following these steps:
1. Create a password protected post/page
1. Check the password input field on the frontend

## Screenshots

If this is a visual change please include screenshots of the change at various screen sizes.

<details>
<summary>Desktop (XL, L) + TABLET</summary>

image

</details>

<details>
<summary>Mobile (Small)</summary>

image

</details>

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

### CODE REVIEW

File: inc/template-functions.php [hunk 1]
```diff
@@ -106,55 +106,55 @@ function twenty_twenty_one_get_the_archive_title() {
 
 	if ( is_tag() ) {
 		return sprintf(
-			/* Translators: %s: The term title. */
+		/* Translators: %s: The term title. */
```
Comment (aristath) [inline inc/template-functions.php]: :thinking: Why was the indentation of these changed?

File: inc/template-functions.php [hunk 2]
```diff
@@ -440,3 +438,22 @@ function twenty_twenty_one_print_first_instance_of_block( $block_name, $content
...
+ *
+ * @since 1.0.0
+ *
+ * @param int|WP_Post $post Optional. Post ID or WP_Post object. Default is global $post.
+ * @return string HTML content for password form for password protected post.
+ */
+function twenty_twenty_one_password_form( $post = 0 ) {
+	$post   = get_post( $post );
+	$label  = 'pwbox-' . ( empty( $post->ID ) ? wp_rand() : $post->ID );
+	$output = '<p>' . __( 'This content is password protected. To view it please enter your password below:', 'twentytwentyone' ) . '</p>
```
Comment (aristath) [inline inc/template-functions.php]: Should probably be escaped

```suggestion
	$output = '<p>' . esc_html__( 'This content is password protected. To view it please enter your password below:', 'twentytwentyone' ) . '</p>
```

File: inc/template-functions.php [hunk 3]
```diff
@@ -440,3 +438,22 @@ function twenty_twenty_one_print_first_instance_of_block( $block_name, $content
...
+	<form action="' . esc_url( site_url( 'wp-login.php?action=postpass', 'login_post' ) ) . '" class="post-password-form" method="post">
+	<label class="post-password-form__label" for="' . $label . '">' . __( 'Password:', 'twentytwentyone' ) . '</label><input class="post-password-form__input" name="post_password" id="' . $label . '" type="password" size="20" /><input type="submit" class="post-password-form__submit" name="Submit" value="' . esc_attr_x( 'Enter', 'post password form', 'twentytwentyone' ) . '" /></form>
```
Comment (aristath) [inline inc/template-functions.php]: Escaping strings & translating `Submit`

```suggestion
	<label class="post-password-form__label" for="' . esc_attr( $label ) . '">' . esc_html__( 'Password:', 'twentytwentyone' ) . '</label><input class="post-password-form__input" name="post_password" id="' . esc_attr( $label ) . '" type="password" size="20" /><input type="submit" class="post-password-form__submit" name="' . esc_attr__( 'Submit', 'twentytwentyone' ) . '" value="' . esc_attr_x( 'Enter', 'post password form', 'twentytwentyone' ) . '" /></form>
```

**Comment by rolf-yoast** [comment]
CR 👍

**Review by aristath — CHANGES_REQUESTED** [review]

File: inc/template-functions.php (hunk 1 above)
Comment (SResok) [reply inc/template-functions.php]: Fault of the formatter

**Review by aristath — APPROVED** [review]

**Pull request merged**